prime-dotnet-webapi/Services/PartyService.cs

```diff
@@ -201,5 +206,40 @@ private IQueryable<Party> GetBasePartyQuery()
                 .Include(p => p.PartyEnrolments)
                 .Include(p => p.PartyCertifications);
         }
+
+        public async Task<int> UpdatePartyHpdid(int limit)
+        {
+            //query party ID where HPDID is null
+            var partyIds = await _context.Parties
+                .Where(p => p.HPDID == null)
+                .OrderBy(p => p.Id)
+                .Select(p => p.Id)
+                .Take(limit)
+                .ToListAsync();
+
+            var counter = 0;
+            foreach (var pId in partyIds)
+            {
+                //for each party ID, get the party record
+                var party = await _context.Parties.Where(p => p.Id == pId).FirstOrDefaultAsync();
+                //get the user object from keycloak
+                var party_user = await _keycloakClient.GetUser(party.UserId);
+                //assign keycloak username to HPDID
+                party.HPDID = party_user.UserName;
+
+                ++counter;
+            }
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                return -1;
```

Comment:
Would `0` be more accurate?

Reply:
if exception was thrown, -1 is a better indication of something went wrong. Anyhow, updated the code to log the error as well.